src/main/src/main/java/org/geoserver/config/GeoServerDataDirectory.java

```diff
@@ -1234,7 +1234,9 @@ else if ( l.getResource() instanceof WMSLayerInfo ) {
             public URL locateResource(String uri) {
                 URL url = super.locateResource(uri);
                 if(url.getProtocol().equalsIgnoreCase("resource")) {
-                    return fileToUrlPreservingCqlTemplates(urlToResource(url).file());
+                    //GEOS-7025: Just get the path; don't try to create the file
+                    return fileToUrlPreservingCqlTemplates(
+                            Paths.toFile(root(), urlToResource(url).path()));
```

Comment:
That was my origional approach as well, but that caused it to throw a very misleading exception down the line.
This way the error message actually reflects the fact that the file is missing.